prov/shm/src/smr_ep.c

```diff
@@ -1329,6 +1333,15 @@ static int smr_ep_ctrl(struct fid *fid, int command, void *arg)
 			}
 		}
 
+#if HAVE_XPMEM
+		if (smr_env.use_xpmem && xpmem)
+			ep->region->xpmem_cap_self = SMR_VMA_CAP_ON;
+		else
+			ep->region->xpmem_cap_self = SMR_VMA_CAP_OFF;
+#else
+		ep->region->xpmem_cap_self = SMR_VMA_CAP_OFF;
+#endif
```

Comment:
The ifdef's here shouldn't be needed.

We should also be able to determine if xpmem is enabled by checking a single variable, not both smr_env.use_xpmem and xpmem.  Ideally, those two values should be aligned.  Are they not?